traffic_ops/traffic_ops_golang/profileparameter/parameterprofilebyname.go

```diff
@@ -22,20 +22,59 @@ package profileparameter
 import (
 	"database/sql"
 	"errors"
+	"github.com/apache/trafficcontrol/lib/go-util"
 	"net/http"
 
 	"github.com/apache/trafficcontrol/lib/go-tc"
 	"github.com/apache/trafficcontrol/traffic_ops/traffic_ops_golang/api"
 )
 
+const API_PROFILES_NAME_NAME_PARAMETERS = "profiles/name/:name/parameters"
+
+func GetProfileNameDeprecated(w http.ResponseWriter, r *http.Request) {
+	getProfileName(w, r, true)
+}
+
 func GetProfileName(w http.ResponseWriter, r *http.Request) {
+	getProfileName(w, r, false)
+}
+
+func deprecationString(deprecated bool) *string {
```

Comment:
just to clean up a bit and get rid of some of the ifs, what are your thoughts on something like this instead?
`func handleLocalErr(w http.ResponseWriter, r *http.Request, tx *sql.Tx, errCode int, userErr error, sysErr error, deprecated bool) {
	if deprecated {
		api.HandleDeprecatedErr(w, r, tx, errCode, userErr, sysErr, util.StrPtr(API_PROFILES_NAME_NAME_PARAMETERS))
	} else {
		api.HandleErr(w, r, tx, errCode, userErr, sysErr)
	}
}`

Reply:
actually it might be good to put that in api.go and pull the alternative out as a parameter as well so that we can use it if we have other deprecated endpoints that are used in their respective alternatives as well

Reply:
I like the idea, that's what I was originally trying to do with my change to HandleDeprecatedErr.